aten/src/ATen/native/cpu/QuickPartition.cpp

```diff
@@ -0,0 +1,572 @@
+      3,4,5,6,0,1,2,7,
+      0,1,2,4,5,6,3,7,
+      1,2,4,5,6,0,3,7,
+      0,2,4,5,6,1,3,7,
+      2,4,5,6,0,1,3,7,
+      0,1,4,5,6,2,3,7,
+      1,4,5,6,0,2,3,7,
+      0,4,5,6,1,2,3,7,
+      4,5,6,0,1,2,3,7,
+      0,1,2,3,5,6,4,7,
+      1,2,3,5,6,0,4,7,
+      0,2,3,5,6,1,4,7,
+      2,3,5,6,0,1,4,7,
+      0,1,3,5,6,2,4,7,
+      1,3,5,6,0,2,4,7,
+      0,3,5,6,1,2,4,7,
+      3,5,6,0,1,2,4,7,
+      0,1,2,5,6,3,4,7,
+      1,2,5,6,0,3,4,7,
+      0,2,5,6,1,3,4,7,
+      2,5,6,0,1,3,4,7,
+      0,1,5,6,2,3,4,7,
+      1,5,6,0,2,3,4,7,
+      0,5,6,1,2,3,4,7,
+      5,6,0,1,2,3,4,7,
+      0,1,2,3,4,6,5,7,
+      1,2,3,4,6,0,5,7,
+      0,2,3,4,6,1,5,7,
+      2,3,4,6,0,1,5,7,
+      0,1,3,4,6,2,5,7,
+      1,3,4,6,0,2,5,7,
+      0,3,4,6,1,2,5,7,
+      3,4,6,0,1,2,5,7,
+      0,1,2,4,6,3,5,7,
+      1,2,4,6,0,3,5,7,
+      0,2,4,6,1,3,5,7,
+      2,4,6,0,1,3,5,7,
+      0,1,4,6,2,3,5,7,
+      1,4,6,0,2,3,5,7,
+      0,4,6,1,2,3,5,7,
+      4,6,0,1,2,3,5,7,
+      0,1,2,3,6,4,5,7,
+      1,2,3,6,0,4,5,7,
+      0,2,3,6,1,4,5,7,
+      2,3,6,0,1,4,5,7,
+      0,1,3,6,2,4,5,7,
+      1,3,6,0,2,4,5,7,
+      0,3,6,1,2,4,5,7,
+      3,6,0,1,2,4,5,7,
+      0,1,2,6,3,4,5,7,
+      1,2,6,0,3,4,5,7,
+      0,2,6,1,3,4,5,7,
+      2,6,0,1,3,4,5,7,
+      0,1,6,2,3,4,5,7,
+      1,6,0,2,3,4,5,7,
+      0,6,1,2,3,4,5,7,
+      6,0,1,2,3,4,5,7,
+      0,1,2,3,4,5,6,7,
+      1,2,3,4,5,0,6,7,
+      0,2,3,4,5,1,6,7,
+      2,3,4,5,0,1,6,7,
+      0,1,3,4,5,2,6,7,
+      1,3,4,5,0,2,6,7,
+      0,3,4,5,1,2,6,7,
+      3,4,5,0,1,2,6,7,
+      0,1,2,4,5,3,6,7,
+      1,2,4,5,0,3,6,7,
+      0,2,4,5,1,3,6,7,
+      2,4,5,0,1,3,6,7,
+      0,1,4,5,2,3,6,7,
+      1,4,5,0,2,3,6,7,
+      0,4,5,1,2,3,6,7,
+      4,5,0,1,2,3,6,7,
+      0,1,2,3,5,4,6,7,
+      1,2,3,5,0,4,6,7,
+      0,2,3,5,1,4,6,7,
+      2,3,5,0,1,4,6,7,
+      0,1,3,5,2,4,6,7,
+      1,3,5,0,2,4,6,7,
+      0,3,5,1,2,4,6,7,
+      3,5,0,1,2,4,6,7,
+      0,1,2,5,3,4,6,7,
+      1,2,5,0,3,4,6,7,
+      0,2,5,1,3,4,6,7,
+      2,5,0,1,3,4,6,7,
+      0,1,5,2,3,4,6,7,
+      1,5,0,2,3,4,6,7,
+      0,5,1,2,3,4,6,7,
+      5,0,1,2,3,4,6,7,
+      0,1,2,3,4,5,6,7,
+      1,2,3,4,0,5,6,7,
+      0,2,3,4,1,5,6,7,
+      2,3,4,0,1,5,6,7,
+      0,1,3,4,2,5,6,7,
+      1,3,4,0,2,5,6,7,
+      0,3,4,1,2,5,6,7,
+      3,4,0,1,2,5,6,7,
+      0,1,2,4,3,5,6,7,
+      1,2,4,0,3,5,6,7,
+      0,2,4,1,3,5,6,7,
+      2,4,0,1,3,5,6,7,
+      0,1,4,2,3,5,6,7,
+      1,4,0,2,3,5,6,7,
+      0,4,1,2,3,5,6,7,
+      4,0,1,2,3,5,6,7,
+      0,1,2,3,4,5,6,7,
+      1,2,3,0,4,5,6,7,
+      0,2,3,1,4,5,6,7,
+      2,3,0,1,4,5,6,7,
+      0,1,3,2,4,5,6,7,
+      1,3,0,2,4,5,6,7,
+      0,3,1,2,4,5,6,7,
+      3,0,1,2,4,5,6,7,
+      0,1,2,3,4,5,6,7,
+      1,2,0,3,4,5,6,7,
+      0,2,1,3,4,5,6,7,
+      2,0,1,3,4,5,6,7,
+      0,1,2,3,4,5,6,7,
+      1,0,2,3,4,5,6,7,
+      0,1,2,3,4,5,6,7,
+      0,1,2,3,4,5,6,7,
+};
+
+template <typename T, bool largest>
+int_same_size_t<T> nan_partition(T *begin, T *end,
```

Comment:
So, uh, what exactly does this do? Just partition away the NaNs? Or something else? A comment would be worth a lot here!

Reply:
Oh, it looks like this is what you call when the pivot is a NaN. That's all I needed in the comment :)

Reply:
OOC: which test exercises this case?